### ISSUE TITLE
Sequence Packing Currently Limited to Micro-Batch Size = 1

### ISSUE THREAD

**Issue by BoxueYang** [EvolvingLMMs-Lab/LLaVA-OneVision-1.5#62]
Hi team! 👋 

First, thank you for all your hard work on this amazing project! I've been experimenting with the sequence packing feature and noticed something that might be worth discussing.

#### Observation
When using sequence packing (via `OFFLINE_PACKED_DATA=1`), the training requires micro-batch size (MBS) to be exactly 1 due to this assertion in `pretrain_llavaov_1_5.py` at line 147:

```python
assert cu_lengths.shape[0] == 1, "micro-batch-size must be 1 for packing"
```

#### Context
I'm training on H200 GPUs (141GB memory) with sequence length 32768, and currently setting MBS=1 leaves quite a bit of memory headroom unused. It would be wonderful if we could leverage multiple micro-batches while still benefiting from sequence packing's memory efficiency.

#### Question
Is this limitation intentional, or would it be feasible to consider supporting MBS > 1 with packed sequences? I imagine this could potentially improve training throughput on high-memory GPUs while maintaining the padding reduction benefits of packing.

#### What I've Noticed
- The current implementation has separate handling for packed versus non-packed data
- The packed sequence parameters appear to be designed for individual samples
- There are differences in how attention masks are processed between the two modes

I'd be happy to help test or provide more details if this is something the team would consider enhancing! Just wanted to bring this to your attention as a potential area for optimization.

Thanks again for your time and this great project! 🚀
